### PR TITLE
John conroy/pdf

### LINKED ISSUE #1886: Enable users to view pdf files within dataset detail page
We want users to be able to better interact with the files without having to download them. As a start, we'd like to display pdf files in a modal.

Notes from @tsliaw for design:
- Use a pdf icon button for users to open the file viewer.
- Add an 'x' button at the top right of the pdf for users to exit the file viewer.
- Display an alert in the event of an error opening the pdf.

### PR DESCRIPTION
**Pull request by john-conroy**
Fix #1886. I recommend checking this out locally. Here's an [example](http://localhost:5001/browse/dataset/53fe8a84564b313d4e8ee90f337129dc). Working on tests.

Screen Shot 2021-08-10 at 3 49 36 PM

Screen Shot 2021-08-10 at 3 49 10 PM

### CODE REVIEW

**Review by mccalluc — CHANGES_REQUESTED** [review]
Little things:
- pseudo code / explanation of the state changes. Seems like a simple finite-state-machine.
- Suppress pagination if it's only one page?
- Design: Maybe incorporate the pagination control into the same rect as the viewer? But I'd defer to @tsliaw.
- If it's a document with text, is there an option to full-screen it?

Big things:
- Can you collect some of the examples of PDFs that don't work? Are they surfacing bugs in the JS rendering, or are they actually corrupt? In the latter, we need to look at where they're coming from, and how to raise a red flag earlier. --> Put on UI agenda, but if you have a few examples handy, slack me.
- There are PDFs which are valid, but take forever to render. [PromotorRatioLogPlot.pdf](https://assets.hubmapconsortium.org/53cc1f9c21907d8f7880eed6828ba8a1/PromotorRatioLogPlot.pdf) for example. Opening these in a new tab isn't great, but trying to render them in single threaded JS is a nightmare: the tab locks up and there's nothing you can do to exit the modal.

So... I have deep misgivings about this, and I'm sorry not to raise them earlier. I'd suggest putting this on hold until we can go through the problems with Nils.

**Review by tsliaw — APPROVED** [review]
UI-wise I think everything looks good.

> Design: Maybe incorporate the pagination control into the same rect as the viewer? 

I think I talked about this with John before, but when the pagination arrows were incorporated with the viewer, it looked odd so I'm fine with how it is as of now.

File: context/app/static/js/components/files/PDFViewer/PDFViewer.jsx
Comment (john-conroy) [reply context/app/static/js/components/files/PDFViewer/PDFViewer.jsx]: Added a comment.

File: context/app/static/js/components/files/PDFViewer/PDFViewer.jsx
Comment (john-conroy) [reply context/app/static/js/components/files/PDFViewer/PDFViewer.jsx]: Fixed the variable names.

**Comment by john-conroy** [comment]
As we discussed the underlying `pdfjs` uses a service worker so it shouldn't block the main thread. I fixed the freezing issue for  `PromotorRatioLogPlot.pdf` by disabling text layer rendering.

**Review by mccalluc — APPROVED** [review]

**Pull request merged**
